tmlibrary/tmlib/tools/classification.py

```diff
@@ -114,3 +169,80 @@ def process_request(self, submission_id, payload):
             self.save_result_values(
                 mapobject_type_name, result_id, predicted_labels
             )
+
+    def save_selections(self, submission_id, payload):
+        '''Processes a client tool request and inserts the generated result
+        into the database. This function deals with the save labels jobs.
+        The `payload` is expected to have the following form::
+
+            {
+                "choosen_object_type": str,
+                "selected_features": [str, ...],
+                "training_classes": [
+                    {
+                        "name": str,
+                        "object_ids": [int, ...],
+                        "color": str
+                    },
+                    ...
+                ],
+                "options": {
+                    "method": str,
+                    "n_fold_cv": int
+                },
+                "task": str (either 'classification' or 'saveLabels'),
+                name: str
+
+            }
+
+        Parameters
+        ----------
+        submission_id: int
+            ID of the corresponding job submission
+        payload: dict
+            description of the tool job
+        '''
+        logger.debug('Saving current selections for submission id: '
+                     + str(submission_id))
+        mapobject_type_name = payload['chosen_object_type']
+
+        labels = dict()
+        label_map = dict()
+        for cls_id, cls in enumerate(payload['training_classes']):
+            labels.update({val: float(cls_id) for val in cls['object_ids']})
+            label_map[float(cls_id)] = {'name': cls['name'],
+                                        'color': cls['color']}
+
+        unique_labels = np.unique(labels.values())
+
+        # Create a MultiIndex pandas.Series for the input labels, because
+        # the save_results_values expects such a pandas Series.
+        # Keys are mapobject ids, values are the actual labels
+        # Hard-coding tpoint = 0 is suboptimal, but I don't know how I
+        # would get the actual tpoint information and as far as I can see, it
+        # does not matter. It's just required to create the same kind of data
+        # structure to be saved as in perform_classification.
+        # To fix this, check how it's done in load_feature_values
+        tpoint = 0
+        indices = [(label, tpoint) for label in labels.keys()]
+
+        index = pd.MultiIndex.from_tuples(
+            indices, names=['mapobject_id', 'tpoint']
+        )
+
+        label_array = np.array(labels.values())
+        label_series = pd.Series(label_array, index=index)
+
+        # Build a name for the result, max. 30 characters in total. Cuts off
+        # input names after character 20. (limited by database settings)
+        label_name = payload['name'][:20] + '-Lbs-' + str(submission_id)
```

Comment:
will this leas to errors in database if names are not unique?

Reply:
No, the actual names always include the submission id and tissuemaps ensures that this is unique. So far, everything was just named "Classification-19" for id 19 etc. Now it's a user-defined name +"-19" for submission id 19

Reply:
I've tested dozens of rounds with varying names, names that are too long, duplicate names etc. and it runs stably

Reply:
Cool, perhaps you could add a warning "(max. 20 chars)" above the box where you write the name

Reply:
Yes, I had already added that :)